modules/vm/main.tf

```diff
@@ -79,4 +79,16 @@ resource "azurerm_linux_virtual_machine" "aap_infrastructure_vm" {
     },
     var.persistent_tags
   )
+
+  # Copy SSH private key file to the installer host to connect to other servers
+  provisioner "file" {
+    connection {
+      type = "ssh"
+      user = "azureuser"
+      host        = azurerm_public_ip.aap_infrastructure_public_ip.ip_address
+      private_key = file(var.infrastructure_admin_ssh_private_key_filepath)
+    }
+    source = "${var.infrastructure_admin_ssh_private_key_filepath}"
+    destination = "/home/azureuser/.ssh/infrastructure_ssh_private_key.pem"
+    }
```

Comment:
We have to copy the private key to only to controller hosts. With this, it will go to in each VM

Reply:
I guess you can copy the loop in https://github.com/ansible-content-lab/aws_terraform_deployment/pull/12/files

Reply:
Updated code to handle the hit or miss issue that was discussed in standup, works as expected now. The issue was that terraform_data resource that has file provisioner was not getting run at times 
now with the addition of triggers, it will always run when new vm's are created